lambdas/src/utils/TheGraphClient.ts

```diff
@@ -163,13 +163,17 @@ export class TheGraphClient {
   }
 
   private async getWearablesByOwner(subgraph: keyof URLs, owner: string) {
-    const query: Query<{ nfts: { urn: string }[] }, WearableId[]> = {
+    const query: Query<
+      { nfts: { urn: string; collection: { isApproved: boolean } }[] },
+      { id: WearableId; isApproved: boolean }[]
+    > = {
       description: 'fetch wearables by owner',
       subgraph: subgraph,
       query: QUERY_WEARABLES_BY_OWNER,
-      mapper: (response) => response.nfts.map(({ urn }) => urn)
+      mapper: (response) => response.nfts.map(({ urn, collection }) => ({ id: urn, isApproved: collection.isApproved }))
     }
-    return this.paginatableQuery(query, { owner: owner.toLowerCase() })
+    const wearables = await this.paginatableQuery(query, { owner: owner.toLowerCase() })
```

Comment:
should we need to filter before paginating?

Reply:
`paginatableQuery` will always retrieve all wearables (approved or not) and we are not paginating the result, so it doesn't make sense for the time being